src/adr/0041-james-servers-offering.md

```diff
@@ -0,0 +1,109 @@
+a default configuration using Derby embedded database, this server have no external dependencies by default.
+However the still large amount of configuration can make new comers confused. It currently satisfies operator 
+looking for self hosting solutions as well as those looking for an extensible server.
+
+ - **JPA SMTP** (`server/container/guice/jpa-smtp`) demonstrate a Mail Transfer Agent, dedicated to mail
+processing, that do not ship the James Mailbox, as well as Mail Delivery protocols like IMAP, JMAP or POP3.
+
+We can notice that this offer is specified in technical terms, and do not directly map to identified user needs.
+
+Also the large amount of servers makes the choice harder to make for newcomers.
+
+## Decision
+
+We should deliver a subset of mail server directly adressing users needs, and brand them according to their use cases
+and not due to their technical details.
+
+In order to get a clearer offer we should first reduce the server cardinality. This can be achieved via:
+
+ - **Spring server** deprecation and removal
+ - Merging Guice servers with their LDAP variations (see [ADR 36 about guice module choosing](0036-against-use-of-conditional-statements-in-guice-modules.md))
+ - **Cassandra-guice** deprecation and removal
+
+We should then rename our servers in order to match the use cases:
+
+ - **Basic server** addresses the self-hosting use case. It will be based on current *jpa-guice* server but with the clear intent to 
+lower adoption barriers. It exposes limited configurations in order not to confuse the user. Configuration and documentation is centered 
+around self hosting use cases.
+
+ - **Advanced server** addresses the customization use cases. It will be based on the current *jpa-guice* server. It differs from the basic server
+as more features will be packaged as part of it, configuration shows via examples advanced usage possibilities, and the documentation is centered 
+around feature extensibility.
+
+Note that transitionning from basic server to the advanced one will be easy as they rely on the same technologies.
+
+ - **Distributed server** will remain unchanged.
+
+ - **Testing server** will rely on current `memory guice` server.
+
+ - **Mail processing server** will rely on current `JPA SMTP guice` server.
+
+Also, we should separate our server application from the technical guice bindings, and group them together for better discoverability.
+
+We should adopt this folder structure:
+
+```
+server/apps
+\_____ basic-server
+\_____ advanced-server
+\_____ distributed-server
+\_____ testing-server
+\_____ mail-processing-server
+```
+
+## Consequences
+
+By better focussing our offer on community needs we expect a larger adoption of James project servers.
+
+Such a split will furthermore ease documentation efforts.
+
+The split between basic and advanced server will furthermore ease bug reports as the customization capabilities of the basic server are limited.
```

Comment:
that's the main disagreement point.

Reply:
Perhaps we need to post a clear hypothesis and somehow test it in order to resolve these kinds of disagreements more objectively.

Reply:
Well, I do think "1 use case = 1 server" is quite nice.

Once we have separate products we can start working on configuration simplifications.

For the self hosting the mailetcontainer.xml file could be replaced with (middle term) : 

```
# (optional)
# smtp.gateway.host=smtpout.my.cloud.supplier
# smtp.gateway.password=xyz

spf.verify=true

dkim.verify=true
dkim.sign=true
dkim.key=/path/to/something

# We can think of other "ready to use features" that would just need simple "properties" configuration to enable
# Let's say: AntiSpam, Virus Scan, etc..
```

(Which focuses on trust issues self hosting users gets). A self hosting user should not even need to know what a mailet / matcher is in order to achieve is basic configuration. Only minimal opt-in knowledge of the email domain is needed.

While for the advanced one we want to keep the full power of mailet-container.

Another example is regarding keystore (SSL management) : 

 - An advanced users might want to specify a SSL certificate per protocol
 - A basic user will for sure want to have a single SSL certificate. From it we can derive all the protocols configuration

Splitting products helps you better address these needs. Keeping a single product don't allow it.

Furthermore, it would enable writing documentation addressing different use cases: self hosting and "extensibility" which, by definition, we don't know what the advanced operator would like to do.